docker.go

```diff
@@ -245,6 +245,14 @@ func (c *DockerContainer) Stop(ctx context.Context, timeout *time.Duration) erro
 
 // Terminate is used to kill the container. It is usually triggered by as defer function.
 func (c *DockerContainer) Terminate(ctx context.Context) error {
+	select {
+	// close reaper if it was created
+	case c.terminationSignal <- true:
+	default:
+	}
+
+	defer c.provider.client.Close()
```

Comment:
What will happen if a potential error happens while closing the client? Will it be swallowed?

Reply:
TBH the whole approach of closing provider (to close client idle connections) in a hundred of places feels wrong.
Maybe there should be a terminal state of the container (after Terminate call?) that would close client and all other methods would return error if they are called after.

Reply:
TBH the whole approach of closing provider (to close client idle connections) in a hundred of places feels wrong.
Maybe there should be a terminal state of the container (after Terminate call?) that would close client and all other methods would return error if they are called after.

Reply:
I'm going to start a discussion for an upcoming V1 release. This kind of discussions are welcome for that thread